### PR TITLE
fix(modal): add useFocusLock hook. fix weird focus states

### PR DESCRIPTION
**Pull request by rcgillet**

### CODE REVIEW

File: packages/react-magma-dom/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import * as React from 'react';
 import styled from '@emotion/styled';
 import { Global, css } from '@emotion/core';
-import {
-  getTrapElements,
-  getTrapElementsAndFocus,
-  getFocusedElementIndex,
-} from './utils';
+// import {
```
Comment (aaronmaturen) [inline packages/react-magma-dom/src/components/Modal/Modal.tsx:4]: delete the comments 😈

File: packages/react-magma-dom/src/hooks/useFocusLock.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
```
Comment (aaronmaturen) [inline packages/react-magma-dom/src/hooks/useFocusLock.ts:1]: should we create a docs page for this? Makes sense that other teams (or components external to dom) would want this too

Comment (aaronmaturen) [reply packages/react-magma-dom/src/hooks/useFocusLock.ts:1]: in the same vein... we should export it in `./src/index.ts`

File: packages/react-magma-dom/src/components/Modal/Modal.tsx (hunk 1 above)
Comment (aaronmaturen) [reply packages/react-magma-dom/src/components/Modal/Modal.tsx:4]: probably delete that code from utils also...

File: packages/react-magma-dom/package.json [hunk 3]
```diff
@@ -50,6 +50,7 @@
   "devDependencies": {
     "@emotion/core": "^10.1.1",
     "@emotion/styled": "^10.0.27",
+    "@testing-library/user-event": "^13.2.0",
```
Comment (aaronmaturen) [inline packages/react-magma-dom/package.json:53]: testing-library is at the root level... and if they're here, versions must match

Comment (aaronmaturen) [reply packages/react-magma-dom/package.json:53]: this is why package-lock showed up

**Review by aaronmaturen — CHANGES_REQUESTED** [review]

**Review by jamiesmyln — APPROVED** [review]
I didn't notice any strange behavior with components that use modals.

**Review by aaronmaturen — APPROVED** [review]

**Pull request merged**